bundles/org.openhab.binding.playstation/src/main/java/org/openhab/binding/playstation/internal/PS4Handler.java

```diff
@@ -0,0 +1,852 @@
+                    updateState(CHANNEL_OSK_TEXT, StringType.valueOf(oskText));
+                    break;
+                case OSK_START_RSP:
+                case OSK_CONTROL_REQ:
+                case COMMENT_VIEWER_START_RESULT:
+                case SCREEN_SHOT_RSP:
+                case APP_START2_RSP:
+                case LOGOUT_RSP:
+                    break;
+                default:
+                    logger.debug("Unknown response, command:{}. Missing case.", command);
+                    break;
+            }
+        } else {
+            logger.debug("Unknown resp-cmd, size:{}, command:{}, status:{}, data:{}.", size, cmdValue, statValue,
+                    respBuff);
+        }
+        return command;
+    }
+
+    private SocketChannel getConnection() throws IOException {
+        return getConnection(true);
+    }
+
+    private SocketChannel getConnection(boolean requiresLogin) throws IOException {
+        SocketChannel channel = null;
+        SocketChannelHandler handler = socketChannelHandler;
+        if (handler == null || !handler.isChannelOpen()) {
+            try {
+                handler = new SocketChannelHandler();
+                socketChannelHandler = handler;
+            } catch (IOException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                throw e;
+            }
+        }
+        channel = handler.getChannel();
+        if (!loggedIn && requiresLogin) {
+            login(channel);
+        }
+        return channel;
+    }
+
+    private void sendPacketToPS4(ByteBuffer packet, SocketChannel channel, boolean encrypted, boolean restartTimeout) {
+        PS4Command cmd = PS4Command.valueOfTag(packet.getInt(4));
+        logger.debug("Sending {} packet.", cmd);
+        try {
+            if (encrypted) {
+                ByteBuffer outPacket = ps4Crypto.encryptPacket(packet);
+                channel.write(outPacket);
+            } else {
+                channel.write(packet);
+            }
+            if (restartTimeout) {
+                setupConnectionTimeout(config.connectionTimeout);
+            }
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    private void sendPacketEncrypted(ByteBuffer packet, SocketChannel channel) {
+        sendPacketToPS4(packet, channel, true, true);
+    }
+
+    private void sendPacketEncrypted(ByteBuffer packet) {
+        sendPacketEncrypted(packet, true);
+    }
+
+    private void sendPacketEncrypted(ByteBuffer packet, boolean requiresLogin) {
+        try {
+            SocketChannel channel = getConnection(requiresLogin);
+            if (requiresLogin && !loggedIn) {
+                scheduler.schedule(() -> sendPacketToPS4(packet, channel, true, requiresLogin), 250,
+                        TimeUnit.MILLISECONDS);
```

Comment:
You should keep track of the returned future so that it can be canceled if this handler gets disposed.